### PR TITLE
MPA-6137-CapsulaHub-UI-controls-library

### PR DESCRIPTION
**Pull request by sergiupadure**

### CODE REVIEW

File: src/lib/buttons/default.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import styled from 'styled-components';
+
+const StyledButton = styled.button`
+  font-size: 1em;
+  font-family: Montserrat;
+  font-weight: 500;
+  text-align: center;
+  background-color: ${props => props.theme.props[0]};
```
Comment (taras-shedenko) [inline src/lib/buttons/default.js]: Let's make props an object (with a defined type?). props[0] - you'll kill yourself in a half of year

File: src/lib/json-editor/json-editor.js [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+  base0A: '#000',
+  base0B: '#DEDEDE',
+  base0C: '#57D7FF',
+  base0D: '#DEDEDE',
+  base0E: '#DEDEDE',
+  base0F: '#DEDEDE'
+};
+
+const Container = styled.div`
+  background: #3F3F3F;
```
Comment (taras-shedenko) [inline src/lib/json-editor/json-editor.js]: Let's put literal values into constants, like this: width ${width100pct}

File: src/lib/buttons/default.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import styled from 'styled-components';
+
+const StyledButton = styled.button`
+  font-size: 1em;
```
[the remark this answers is not in the document]
Comment (sergiupadure) [reply src/lib/buttons/default.js]: @95kach what do you think about extracting style values in a separate file? common for the whole library?

File: src/lib/buttons/default.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import styled from 'styled-components';
+
+const StyledButton = styled.button`
```
[the remark this answers is not in the document]
Comment (sergiupadure) [reply src/lib/buttons/default.js]: Renamed it to "CButton", but let's keep thinking on better naming scheme

File: src/lib/buttons/default.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  &:hover {
+    background-color: ${props => props.theme.props[1]};
+  }
+
+  ${props => props.css};
+`;
+
+//theme name: background color, hover background color, selected background color, text color, border
+const themes = [
+  {name: 'default', props: ['#57D7FF', '#4EBEE0', '#3B3B3B', '#57D7FF']},
```
[the remark this answers is not in the document]
Comment (sergiupadure) [reply src/lib/buttons/default.js]: Done

File: src/lib/buttons/default.js (hunk 1 above)
Comment (sergiupadure) [reply src/lib/buttons/default.js]: :D agree, done.

File: package.json [hunk 6]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "capsulahub-ui",
+  "version": "0.0.1",
+  "description": "CapsulaJS UI component library",
+  "main": "./dist/main.js",
+  "scripts": {
+    "start": "parcel src/docs/index.html",
```
[the remark this answers is not in the document]
Comment (95kach) [reply package.json]: it simpler to use parcel for development.
Btw. You won't run dev server without `webpack-dev-server` with separate config etc.
Better to use parcel.

File: .gitignore [hunk 7]
```diff
@@ -0,0 +1,9 @@
+.npmrc
+package-lock.json
```
Comment (taras-shedenko) [inline .gitignore]: This file should be commited

File: package.json [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    "build": "webpack --mode production --progress --display-error-details"
+  },
+  "dependencies": {
+    "clean-webpack-plugin": "^0.1.19",
+    "react": "^16.2.0",
```
Comment (taras-shedenko) [inline package.json]: Whar about latest React?

File: package.json [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+    "react-ace": "^6.1.4",
+    "react-click-outside": "^3.0.1",
+    "react-dom": "^16.2.0",
+    "react-json-view": "^1.19.1",
+    "typeface-montserrat": "^0.0.54"
+  },
+  "devDependencies": {
+    "babel-core": "^6.26.3",
```
Comment (taras-shedenko) [inline package.json]: What about Babel 7?

File: src/lib/form/button.js [hunk 10]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import styled from 'styled-components';
+
+const CButton = styled.button`
+  font-size: 1em;
```
Comment (taras-shedenko) [inline src/lib/form/button.js]: Replace literals with named constants

File: src/lib/form/dropdown.js [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+  select(selected) {
+    this.setState({ selected });
+    this.toggle();
+  }
+  
+  render() {
+    const { title,items, selected, isOpen } = this.state;
+    
+    return (<Container>
+      <Header onClick={() => this.toggle()}>
```
Comment (taras-shedenko) [inline src/lib/form/dropdown.js]: Move function definition out of render method

File: package.json (hunk 9 above)
Comment (95kach) [reply package.json]: Do we really need support async / await or another newest features ?? just for the package with a few components?

You will get a lot of pain. I already tried to do it.

File: package.json (hunk 8 above)
Comment (95kach) [reply package.json]: done

File: src/lib/form/dropdown.js [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+        <Title>{Number.isInteger(selected) ? items[selected].label : title}</Title>
+        {isOpen ? <ArrowUp/> : <ArrowDown/>}
+      </Header>
+      {this.state.isOpen &&
```
Comment (taras-shedenko) [inline src/lib/form/dropdown.js]: use isOpen instead of this.state.isOpen

File: .gitignore (hunk 7 above)
Comment (95kach) [reply .gitignore]: done

File: src/lib/form/dropdown.js [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+        <List>
+          {items.map((item, index) => (<Item key={index} onClick={() => this.select(index)}>{item.label}</Item>))}
+        </List>
+      }
+    </Container>);
+  }
+}
+
+export { Dropdown };
```
Comment (taras-shedenko) [inline src/lib/form/dropdown.js]: Add export statement to the class definition

File: src/lib/form/input.js [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+  &:focus {
+    outline: none;
+  }
+  
+  ::placeholder {
+    color: #B1B1B1;
+  }
+`;
+
+const Input = ({ onChange, placeholder }) => <CInput placeholder={placeholder}
```
Comment (taras-shedenko) [inline src/lib/form/input.js]: Use export here

File: src/lib/form/json-input.js [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+import styled from 'styled-components';
+import { defaultFontFamily } from '../constants';
+import image from '../../assets/settings.png';
+import AceEditor from 'react-ace';
+import 'brace/mode/json';
+import '../theme';
+
+const Container = styled.div`
+  font-family: ${defaultFontFamily};
+  font-style: regular;
```
Comment (taras-shedenko) [inline src/lib/form/json-input.js]: Use named constants instead of literals

File: src/lib/form/button.js (hunk 10 above)
Comment (95kach) [reply src/lib/form/button.js]: done

File: package.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (taras-shedenko) [reply package.json]: Okay

**Pull request merged**